src/modules/python/pbs_hooks/PBS_cray_jacs.PY

```diff
@@ -0,0 +1,339 @@
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+"""
+PBS Professional hook for consuming the Shasta Northbound API.
+
+This hook services the following events:
+- execjob_begin
+- execjob_end
+"""
+
+
+import sys
+import os
+import json as JSON
+import urllib
+import signal
+import time
+import site
+import requests
+import requests_unixsocket
+import pbs
+import pwd
+import copy
+
+requests_unixsocket.monkeypatch()
+
+# ============================================================================
+# Utility functions
+# ============================================================================
+
+
+class OfflineError(Exception):
+    """
+    Exception that will offline the node and reject the event
+    """
+    pass
+
+
+class RejectError(Exception):
+    """
+    Exception that will reject the event
+    """
+    pass
+
+
+class HookHelper(object):
+    """
+    Helper to load config and event
+    """
+    config = None
+    event = None
+
+    def __init__(self):
+        raise Exception('Access class via static methods')
+
+    @classmethod
+    def load_config(cls):
+        """
+        Read the config file
+        """
+        log_function_name()
+        defaults = {
+            'post_timeout': 30,
+            'delete_timeout': 30,
+            'unix_socket_file': '/var/run/jacsd/jacsd.sock',
+        }
+        constants = {
+            'version_uri': '/rm/v1',
+            'resources': {
+                'job': '/jobs',
+                'task': '/tasks'
+            }
+        }
+        # Identify the config file and read in the data
+        config_file = ''
+        if 'PBS_HOOK_CONFIG_FILE' in os.environ:
+            config_file = os.environ['PBS_HOOK_CONFIG_FILE']
+        else:
+            raise RuntimeError('%s: No config file set' % caller_name())
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'config file is %s' % config_file)
+        try:
+            with open(config_file, 'r') as cfg:
+                config = merge_dict(defaults, JSON.load(cfg))
+        except IOError:
+            raise IOError('I/O error reading config file')
+        config = merge_dict(config, constants)
+        pbs.logmsg(pbs.EVENT_DEBUG2, 'loaded config is: %s' % (str(config)))
+        cls.config = config
+
+    @classmethod
+    def get_event(cls):
+        """
+        Fetch the event if it hasn't already been fetched.
+        Return the event
+        """
+        if not cls.event:
+            cls.event = pbs.event()
+        return cls.event
```

Comment:
@bayucan can correct me, but I don't think you are saving anything by caching the event type.  It is a variable that is part of the pbs object.  You're just copying one variable into another.

Reply:
You're right, pbs.event() just returns a PyObject. I've removed it.